### PR TITLE
fix: replace MUI with ui-core for dimension modals (TECH-318, TECH-319)

### PR DESCRIPTION
**Pull request by martinkrulltott**
No functional changes.
Replaces MUI elements with their ui-core counterparts for the following tasks:
https://jira.dhis2.org/browse/TECH-318 - Modal
https://jira.dhis2.org/browse/TECH-319 - Add to layout button

Plus multiple minor updates across various files to support the component updates imposed by https://github.com/dhis2/analytics/pull/382

Upgrades @dhis2/ui-core dep to latest, 4.16.0
Please note that `yarn.lock` has been updated to include `"@dhis2/ui-core@^4.16.0", "@dhis2/ui-core@^4.6.1":`, which fixes an issue with modals and select elements (https://github.com/dhis2/ui/issues/15#issuecomment-607206361)

**Note: Depends on https://github.com/dhis2/analytics/pull/382**

-------------------

Affects all the dimension modals, i.e. _Data_, _Period_, _Org unit_ and dynamic dimensions (_Area_ as example below).

image

image

image

image

### CODE REVIEW

File: packages/app/src/components/DimensionsPanel/Dialogs/AddToLayoutButton/AddToLayoutButton.js [hunk 1]
```diff
@@ -1,20 +1,16 @@
...
+import { SplitButton, Menu, MenuItem, Button } from '@dhis2/ui-core'
 
-import Menu from './Menu'
 import { sGetUiActiveModalDialog, sGetUiType } from '../../../../reducers/ui'
 import { acAddUiLayoutDimensions } from '../../../../actions/ui'
 
-import styles from './styles/AddToLayoutButton.style'
 import i18n from '@dhis2/d2-i18n'
 
 export class AddToLayoutButton extends Component {
```
Comment (edoardo) [inline packages/app/src/components/DimensionsPanel/Dialogs/AddToLayoutButton/AddToLayoutButton.js:16]: We can convert this into a functional component.
Apparently we don't need to use state as `onClose` and `onToggle` are not used anymore?

Comment (martinkrulltott) [reply packages/app/src/components/DimensionsPanel/Dialogs/AddToLayoutButton/AddToLayoutButton.js:16]: Yes, you're right, `onClose` and `onToggle` aren't used anymore. And all components touched by this PR should be converted to functional with css modules 👍

Comment (martinkrulltott) [reply packages/app/src/components/DimensionsPanel/Dialogs/AddToLayoutButton/AddToLayoutButton.js]: Fixed 🚀

**Comment by jenniferarnesen** [comment]
Unfortunately, the dialog change won't work with this implementation. The Dialog content contains MUI components that mess up the layout:

image


If you look in the @dhis2/analytics components (e.g., https://github.com/dhis2/analytics/blob/master/src/components/DataDimension/DataDimension.js#L262), you'll see it. The rendered output becomes messed up:

image

File: packages/app/src/components/DimensionsPanel/Dialogs/DialogManager.js [hunk 2]
```diff
@@ -322,23 +319,44 @@ export class DialogManager extends Component {
...
+                        large
+                    >
+                        <ModalTitle>{dialogTitle}</ModalTitle>
+                        <ModalContent>
+                            {this.renderDialogContent()}
+                        </ModalContent>
+                        <ModalActions>
+                            <ButtonStrip>
+                                <HideButton onClick={this.closeDialog} />
+                                {dialogId && this.renderPrimaryButton(dialogId)}
```
Comment (edoardo) [inline packages/app/src/components/DimensionsPanel/Dialogs/DialogManager.js:354]: Is this `dialogId` check necessary?
It should not happen that `dialogId` is `undefined` at this point...

File: packages/app/src/components/DimensionsPanel/Dialogs/__tests__/DialogManager.spec.js [hunk 3]
```diff
@@ -153,7 +153,8 @@ describe('The DialogManager component', () => {
         )
     })
 
-    it('calls the closeDialog function', () => {
+    // TODO: Fix this failing test
+    it.skip('calls the closeDialog function', () => {
```
Comment (edoardo) [inline packages/app/src/components/DimensionsPanel/Dialogs/__tests__/DialogManager.spec.js:157]: Are you going to have a different test for this or should it be removed?

File: packages/app/src/components/Visualization/StartScreen.js [hunk 4]
```diff
@@ -1,17 +1,18 @@
 import React, { useEffect, useState } from 'react'
 import { connect } from 'react-redux'
 import i18n from '@dhis2/d2-i18n'
+import PropTypes from 'prop-types'
+import { withStyles } from '@material-ui/core/styles'
```
Comment (edoardo) [inline packages/app/src/components/Visualization/StartScreen.js:5]: Can this be also converted or is it out of scope for this PR?

Comment (martinkrulltott) [reply packages/app/src/components/Visualization/StartScreen.js:5]: Haha whoops, just my OCD that forced me to reorder the imports 😄 but there's a separate PR for this coming

File: packages/app/src/components/DimensionsPanel/Dialogs/DialogManager.js (hunk 2 above)
Comment (martinkrulltott) [reply packages/app/src/components/DimensionsPanel/Dialogs/DialogManager.js]: Good catch, just legacy stuff I suppose. Removed it and consolidated the `renderPrimaryButton` fn in to the code as well.

**Comment by janhenrikoverland** [comment]
> image

Add space between action buttons

File: packages/app/src/components/DimensionsPanel/Dialogs/__tests__/DialogManager.spec.js (hunk 3 above)
Comment (martinkrulltott) [reply packages/app/src/components/DimensionsPanel/Dialogs/__tests__/DialogManager.spec.js]: Fixed!

**Pull request merged**